### ISSUE TITLE
[💡SUG] Add weight_decay

### ISSUE THREAD

**Issue by rowedenny** [RUCAIBox/RecBole#655]
**Is your feature request related to a problem? Please describe.**
Look at the current implementation, some basic models do not take the L2 regularization into consideration. 
When our recommender model somehow embeds with self-implemented models, the current RegLoss may not be easy to specify all the learnable weights. 

**Describe the solution you'd like**
Add weight decay to the optimizer

**Describe alternatives you've considered**
In default, we can set the value to be 0.0, and optionally one can add explicit regularization term.

**Additional context**
I launch a PR, but it fails, any reason why? https://github.com/RUCAIBox/RecBole/pull/654

**Comment by rowedenny** [comment]
Got you! Thanks for the response. 

The forum you refer to is interesting, I will take a close look at it. To my best knowledge, a large number of the CV models, typically implemented with BN, are trained with weight_decay yet, yet I barely heard of the failures. Yet it is true that we could be cautious about it.

**Comment by 2017pxy** [comment]
Hi! @rowedenny We have create a new PR(#661 ) to add weight_decay into RecBole referring your idea in #654, and thanks for your contribution to RecBole.

**Comment by rowedenny** [comment]
My pleasure.
